Source/GSLayoutManager.m

```diff
@@ -3218,8 +3223,10 @@ - (void) insertGlyphs: (const NSGlyph*)glyph_list
   glyph_run_t *run;
   int i;
   unsigned int gpos, cpos;
-  NSSize advances[length];
-
+  // TESTPLANT-MAL-07312017: Would be nice to optimize this...
+  //     but we've been seeing stack overwrite crashes due to this...
+  NSSize *advances = NSZoneMalloc(NSDefaultMallocZone(), sizeof(NSSize)*length);
+  
```

Comment:
This looks like it's working around a stack smashing vulnerability, by allocating `advances` onto its own page so you quietly write off the end of it. I'd suggest that there's a problem here in how `advances` is used, probably in `-insertGlyphs:withAdvancements:length:forStartingGlyphAtIndex:characterIndex:` that this change papers over.

Reply:
I looked at that other method today; it only reads the advances and doesn't write them. So there isn't a stack smasher after all, which is good :). I guess what the comment about "stack overwrite crashes" means is that if `length` is too big, then the array is too big to allocate on the stack. Then heap allocation is the right approach.